packages/curve-ui-kit/src/features/slippage-settings/ui/SlippageSettings.tsx

```diff
@@ -1,33 +1,49 @@
 import { ReactNode } from 'react'
 import TuneIcon from '@mui/icons-material/Tune'
-import { IconButton } from '@mui/material'
+import IconButton, { type IconButtonProps } from '@mui/material/IconButton'
 import { useSwitch } from '@ui-kit/hooks/useSwitch'
 import {
   SlippageSettingsModal,
   type SlippageSettingsCallbacks,
   type SlippageSettingsProps,
 } from './SlippageSettingsModal'
 
-type Props = Pick<SlippageSettingsProps, 'maxSlippage'> &
+type SlippageSettingsBaseProps = Pick<SlippageSettingsProps, 'maxSlippage'> &
   Pick<SlippageSettingsCallbacks, 'onSave'> & {
-    /** Custom button component to render instead of the default IconButton */
-    button?: (props: { maxSlippage?: string; onClick: () => void }) => ReactNode
-    /** Custom icon to use in the default IconButton (ignored if button prop is provided) */
-    buttonIcon?: ReactNode
     /** Whether the button should be disabled */
     disabled?: boolean
   }
 
-export const SlippageSettings = ({ disabled = false, button, buttonIcon, maxSlippage, onSave }: Props) => {
+type SlippageSettingsPropsWithButton = SlippageSettingsBaseProps & {
```

Comment:
Personally I'd prefer to make the button property type conditional rather than the entire component prop:

```ts
type Button = {
  onClick: (props: { maxSlippage?: string; onClick: () => void }) => ReactNode
} | {
  icon?: ReactNode
  size?: IconButtonProps['size']
}

type SlippageSettingsProps = {
  button?: Button
}
```

It just doesn't sit well with me to negate props with `never` 🤓